_gsocblogs/2024/blog_Key4hep_BraulioRivas.md

```diff
@@ -0,0 +1,131 @@
+
+**Visualizing**
+
+Having the data loaded on memory, I was able to later develop the visualization of these new datatypes. For each new
+datatype, a few properties have to be shown for an "object" belonging to that collection, and the whole collection has to be
+placed around in a way its easier to understand. I discussed with mentors [what information](https://github.com/key4hep/eede/blob/main/js/types/objects.js) to show and most importantly,
+**how** to visualize them, including relations. After throwing some ideas, I defined a set of common ["views"](https://github.com/key4hep/eede/tree/main/js/views/templates) that repeated across datatypes. So for
+example, a `Cluster`, `Track` or `ReconstructedParticle` collection will be shown as a tree, connected to other objects of the
+same type. But, if a `MCParticle` is related to a `ReconstructedParticle`, it will be shown as a large list going downwards.
+
+| Pull Request                               |                   PR Number                    | Description                                                                                                                     |
+| :----------------------------------------- | :--------------------------------------------: | :------------------------------------------------------------------------------------------------------------------------------ |
+| Visualization of MCParticle + RecoParticle | [#43](https://github.com/key4hep/eede/pull/43) | Views for `RecoParticle` and existing relations between this and `MCParticle`.                                                  |
+| Views for cluster and track                | [#48](https://github.com/key4hep/eede/pull/48) | I added a views for `Cluster` and `Track` plus relations that exists between `MCParticles`-`Cluster` and `MCParticles`-`Track`. |
+| Initial views for particleID and vertex    | [#55](https://github.com/key4hep/eede/pull/55) | More views for `ParticleId` and `Vertex`.                                                                                       |
+
+**Pixi**
+
+eede is a static website. So we are using plain Javascript, HTML and CSS. But for visualization, we were using the Canvas API,
+so all datatypes are drawn in a white canvas. Nevertheless, when having many objects in a same view, it was almost impossible
+to scroll or drag objects around. So I [analyzed](https://github.com/key4hep/eede/pull/63) some of the weak points at that moment.
+I came to the conclusion to change Canvas API to some more performant tool, and to find a way to render objects more efficiently.
+After some research, I found [pixi.js](https://pixijs.com/), a library (which has a minified `js` version) that uses WebGL
+to render any kind of content on a canvas very, but very fast. After a bit of rework, I finally transferred everything to pixi.
```

Comment:
```suggestion
to render any kind of content on a canvas very fast. After a bit of rework, I finally transferred everything to pixi.
```
I think. Otherwise some half sentence is missing here before the `but`